learn/pkg/learn/question.go

```diff
@@ -455,6 +543,23 @@ func (m *QuestionModel) buildAnswerChoicesField(block markdown.Block) error {
 	return nil
 }
 
+func (m *QuestionModel) buildAnswerTextField(block markdown.Block) error {
+	if m.AnswerText != nil || m.Question == nil {
```

Comment:
:pineapple: I am finding all these checks of `m.AnswerText`, `m.AnswerChoices`, `m.Question` against `nil` (either nil or not nil) somewhat confusing. There seems to be a bit of a state machine here and it's a little hard to understand the progression exactly.

Reply:
ok, i just cannot not respond to a 🍍 .
I think you have point. The comparison doesn't belong here and shows again that this code needs a holistic revision when done.
I've hoisted it up, which should make it clearer hopefully.